client/my-sites/themes/atomic-transfer-dialog.tsx

```diff
@@ -93,16 +91,23 @@ class AtomicTransferDialog extends Component< AtomicTransferDialogProps > {
 	}
 
 	continueToActivate() {
-		const { siteId, theme, dispatchActivateTheme, dispatchAcceptAtomicTransferDialog } = this.props;
+		const {
+			siteId,
+			theme,
+			dispatchActivateTheme,
+			dispatchAcceptAtomicTransferDialog,
+			dispatchAcceptActivationModal,
+		} = this.props;
 		if ( siteId ) {
 			dispatchAcceptAtomicTransferDialog( theme.id );
+			dispatchAcceptActivationModal( theme.id );
 			dispatchActivateTheme( theme.id, siteId );
 		}
 	}
 
 	componentDidUpdate( prevProps: Readonly< AtomicTransferDialogProps > ): void {
-		const { siteId, siteSlug, isTransferred, uploadError } = this.props;
-		if ( siteId && siteSlug && prevProps.isTransferred !== isTransferred && isTransferred ) {
+		const { siteId, siteSlug, uploadError, isJetpack } = this.props;
+		if ( siteId && siteSlug && prevProps.isJetpack !== isJetpack ) {
```

Comment:
The site might still be a non-jetpack site after the atomic transfer because the client-side data was not updated. As a result, check whether the site becomes a Jetpack site or not to continue the activation.

Reply:
> because the client-side data was not updated

What would be the cause of this?

Reply:
The cause is when you activate a theme, it will think the site is still not a Jetpack site and then activate the theme immediately. However, the site is transferred so the theme may not be available without installation.